src/docs/arch-tiered-storage.dox

```diff
@@ -0,0 +1,440 @@
+/*! @arch_page arch-tiered-storage Tiered Storage
+
+\warning Tiered storage is considered an experimental part of the WiredTiger
+library, and is not yet ready for production workloads.
+
+@section ts_intro Introduction and Definitions
+
+Tiered storage allows B-Trees to be stored into multiple places, more
+recently updated blocks on local disk and less recently updated blocks in cloud storage.
+The term \a object refers to one of these written
+btree parts, whether the object resides on the local disk, as an
+<em>object file</em>, or in the cloud, as a <em>cloud object</em>.
+The mechanism to create new objects is a WT_SESSION::checkpoint API call
+with the \c flush_tier configuration set.
+For brevity, is often called a \c flush_tier call.
+
+When in use, a tiered btree, like a non-tiered btree, may have some
+recently used or modified data that resides in memory pages.
+This in-memory representation is the same between a tiered btree
+and a non-tiered btree, it is only how data is stored on disk and in
+cloud objects that makes these btrees different.
+
+Tiered storage is configured and enabled on the connection
+via ::wiredtiger_open configuration options
+(e.g. <code>tiered_storage=(name="s3_store"),bucket="..."</code> ).
+Once enabled on the connection, the configuration applies to all tables created.
+The configuration can be overridden on individual tables can be changed with
+configuration options for WT_SESSION::create .
+This allows the caller to specify a different storage provider or bucket name or
+to have the table opt out of tiered storage.
+
+@section ts_objectid Object IDs
+
+In response to a \c flush_tier call, WiredTiger creates a new object
+for each changed btree.  Each new object created gets a new \a
+objectid, incremented from the previous objectid used with that btree.
+The new object lives initially as a regular local disk <em>object file</em>,
+and the disk file name includes the name of the table and the objectid.  This makes
+it is easy to see from a directory listing which tables and which
+objects for that table are represented on disk. Once a new (\a N+1)
+object is created, all writes to the previous (\a N) object are
+completed and that \a N object becomes read-only, like all objects
+before it.  At this point, the \a N object is queued to be copied to
+cloud storage.  After that copy takes place, the local copy of \a N is
```

Comment:
I suggest replace "takes place" to "successfully completes".

Reply:
Fixed.